pkg/adaptor/mongodb/reader.go

```diff
@@ -119,7 +123,7 @@ func (r *Reader) iterateCollection(mgoSession *mgo.Session, in <-chan string, do
 					}
 					iter.Close()
 					s.Close()
-					return
+					break
```

Comment:
So in this case, the query didn't error. We break and wait for a new collection to iterate or the channel to close.. that seems correct. Good catch.